File: org.palladiosimulator.blockchainsystems.core/src/main/java/org/palladiosimulator/blockchainsystems/core/system/BlockchainSystemNodeContextImpl.java
package org.palladiosimulator.blockchainsystems.core.system;

import org.palladiosimulator.blockchainsystems.core.system.abstractions.*;

/**
 * The {@code BlockchainSystemNodeContextImpl} class is an implementation of the {@code BlockchainSystemNodeContext} interface.
 * It is a container that provides access to all components of a blockchain system node.
 *
 * @author Yannik Sproll
 */
public class BlockchainSystemNodeContextImpl implements BlockchainSystemNodeContext {

    private final String _id;
    private final PropagationStrategy<Transaction> _transactionPropagationStrategy;
    private final PropagationStrategy<Block> _blockPropagationStrategy;
    private final NodeP2PNetworkInterface _networkInterface;
    private final MiningProcess _miningProcess;
    private final Blockchain _blockchain;
    private final BlockValidator _blockValidator;
    private final OrphanBlockPool _orphanBlockPool;
    private final BlockFactory _blockFactory;
    private final TransactionFactory _transactionFactory;
    private final TransactionPool _transactionPool;

    public BlockchainSystemNodeContextImpl(
            String id,
            PropagationStrategy<Transaction> transactionPropagationStrategy,
            PropagationStrategy<Block> blockPropagationStrategy,
            NodeP2PNetworkInterface networkInterface,
            MiningProcess miningProcess,
            Blockchain blockchain,
            BlockValidator blockValidator,
            OrphanBlockPool orphanBlockPool,
            BlockFactory blockFactory,
            TransactionFactory transactionFactory,
            TransactionPool transactionPool
    ) {
        _id = id;
        _transactionPropagationStrategy = transactionPropagationStrategy;
        _blockPropagationStrategy = blockPropagationStrategy;
        _networkInterface = networkInterface;
        _miningProcess = miningProcess;
        _blockchain = blockchain;
        _blockValidator = blockValidator;
        _blockFactory = blockFactory;
        _orphanBlockPool = orphanBlockPool;
        _transactionFactory = transactionFactory;
        _transactionPool = transactionPool;
    }

    @Override
    public String getId() {
        return _id;
    }

    @Override
    public PropagationStrategy<Transaction> getTransactionPropagationStrategy() {
        return _transactionPropagationStrategy;
    }

    @Override
    public PropagationStrategy<Block> getBlockPropagationStrategy() {
        return _blockPropagationStrategy;
    }

    @Override
    public NodeP2PNetworkInterface getNetworkInterface() {
        return _networkInterface;
    }

    @Override
    public MiningProcess getMiningProcess() {
        return _miningProcess;
    }

    @Override
    public Blockchain getBlockchain() {
        return _blockchain;
    }

    @Override
    public OrphanBlockPool getOrphanBlockPool() {
        return _orphanBlockPool;
    }

    @Override
    public BlockFactory getBlockFactory() {
        return _blockFactory;
    }

    @Override
    public BlockValidator getBlockValidator() {
        return _blockValidator;
    }

    @Override
    public TransactionFactory getTransactionFactory() {
        return _transactionFactory;
    }

    @Override
    public TransactionPool getTransactionPool() {
        return _transactionPool;
    }
}

File: org.palladiosimulator.blockchainsystems.core/src/main/kotlin/org/palladiosimulator/blockchainsystems/core/system/BlockchainSystemNodeContextImpl.kt
package org.palladiosimulator.blockchainsystems.core.system

import org.palladiosimulator.blockchainsystems.core.geography.GeographicalRegion
import org.palladiosimulator.blockchainsystems.core.system.abstractions.*

/**
 * The [BlockchainSystemNodeContextImpl] class is an implementation of the [BlockchainSystemNodeContext] interface.
 * It is a container that provides access to all components of a blockchain system node.
 *
 * @author Yannik Sproll, Davis Riedel
 */
class BlockchainSystemNodeContextImpl(
  private val id: String,
  private val blockPropagationStrategy: PropagationStrategy<Block>,
  private val transactionPropagationStrategy: PropagationStrategy<Transaction>,
  private val networkInterface: NodeP2PNetworkInterface,
  private val miningProcess: MiningProcess,
  private val blockchain: Blockchain,
  private val blockValidator: BlockValidator,
  private val orphanBlockPool: OrphanBlockPool,
  private val blockFactory: BlockFactory,
  private val geographicalRegion: GeographicalRegion
) : BlockchainSystemNodeContext {
  override fun getId() = id
  override fun getBlockPropagationStrategy() = blockPropagationStrategy
  override fun getTransactionPropagationStrategy() = transactionPropagationStrategy
  override fun getNetworkInterface() = networkInterface
  override fun getMiningProcess() = miningProcess
  override fun getBlockchain() = blockchain
  override fun getOrphanBlockPool() = orphanBlockPool
  override fun getBlockFactory() = blockFactory
  override fun getBlockValidator() = blockValidator
  override fun getGeographicalRegion() = geographicalRegion
}